src/framework/ui/menu/menu.component.tsx

```diff
@@ -0,0 +1,264 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import React from 'react';
+import {
+  ViewProps,
+  ListRenderItemInfo,
+  GestureResponderEvent,
+  View,
+} from 'react-native';
+import {
+  styled,
+  StyledComponentProps,
+  StyleType,
+} from '@kitten/theme';
+import {
+  List,
+  ListProps,
+} from '../list/list.component';
+import {
+  MenuItem,
+  MenuItemType,
+  MenuItemElement,
+  MenuItemProps,
+} from './menuItem.component';
+import { MenuGroup } from './menuGroup.component';
+import { MenuService } from './menu.service';
+
+interface ComponentProps {
+  selectedIndex?: number;
+  onSelect: (index: number, event?: GestureResponderEvent) => void;
+}
+
+export type MenuProps = StyledComponentProps & ComponentProps & Omit<ListProps, 'renderItem'>;
+export type MenuElement = React.ReactElement<MenuProps>;
+
+/**
+ * `Menu` renders vertical list of `MenuItems`.
+ *
+ * @extends React.Component
+ *
+ * @property {string} appearance - Determines the appearance of the component.
+ * Can be `default` or `divider`.
+ * Default is `default`.
+ *
+ * @property {MenuItemType[]} data - Determines menu items.
+ *
+ * @property {number} selectedIndex - The index of selected item.
+ *
+ * @property {(index: number, event?: GestureResponderEvent) => void} onSelect - Fires when
+ * selected item is changed.
+ *
+ * @property Omit<ListProps, 'renderItem'>
+ *
+ * @property StyledComponentProps
+ *
+ * @overview-example Simple Usage
+ *
+ * ```
+ * import React from 'react';
+ * import {
+ *   Menu,
+ *   MenuItemType,
+ * } from 'react-native-ui-kitten';
+ *
+ * interface State {
+ *   selectedIndex: number;
+ * }
+ *
+ * export class MenuShowcase extends React.Component<any, State> {
+ *
+ *   public state: State = {
+ *     selectedIndex: null,
+ *   };
+ *
+ *   private data: MenuItemType[] = [
+ *     { title: 'Item 1' },
+ *     { title: 'Item 2' },
+ *     { title: 'Item 3' },
+ *   ];
+ *
+ *   private onSelect = (selectedIndex: number): void => {
+ *     this.setState({ selectedIndex });
+ *   };
+ *
+ *   public render(): React.ReactNode {
+ *     return (
+ *       <Menu
+ *         data={this.items}
+ *         selectedItem={this.state.selectedIndex}
+ *         onSelect={this.onItemSelect}
+ *       />
+ *     );
+ *   }
+ * }
+ * ```
+ *
+ * @overview-example Eva Styling
+ *
+ * ```
+ * // use code from the example above
+ *
+ * <Menu
+ *   appearance='divider'
+ *   data={this.items}
+ *   selectedItem={this.state.selectedIndex}
+ *   onSelect={this.onItemSelect}
+ * />
+ * ```
+ *
+ * @example Disabled Item
+ *
+ * ```
+ * // use code from example above
```

Comment:
Let's use it :)

Reply:
I guess that just for replacing object props of MenuItemType we don't need the separate full example :)

Reply:
Anyway done.